Adding new tribal bedding type and full compatiblity for VFEV

jupiterlily opened the pull request:
- Created Blanket Defs for Single and double fur beds. Textures to come shortly.

- Added Bedding Defs for new tribal beddings. Still using original textures for now.
    - Changed stuff used to be only leather.
    - Made it 20% less work to make.
    - Set research prerequisite to carpets.
   
- Patches
    - copied function extension from other patches. Not sure if it needs anything else doing to it.
    - Pointed Graphics to textures that don't exist yet. The paths should exist.
    - Hopefully made it so that if Vanilla Furniture Expanded is loaded the research prerequisite for making double bedding is Basic Furniture. This hopefully reduces dependencies. **Please check this, I got very confused with using patch operation sequence and using patchoperation find mod together**
    - Added bedding settings
    - Changed Bed Rest Effectiveness

jupiterlily committed:
Adding new tribal bedding, and modifying VFEV Bed to fit

Created Blanket Defs for Single and double fur beds. Textures to come shortly.

Added Bedding Defs for new tribal beddings. Still using original textures for now.

Changed stuff used to be only leather.
Made it 20% less work to make.

Set research prerequisite to carpets.

Patches

copied function extension from other patches. Not sure if it needs anything else doing to it.

Pointed Graphics to textures that don't exist yet. The paths should exist.

Hopefully made it so that if Vanilla Furniture Expanded is loaded the research prerequisite for making double bedding is Basic Furniture. This hopefully reduces dependencies.

Added bedding settings

Changed Bed Rest Effectiveness

jupiterlily committed:
Merge branch 'master' of https://github.com/jupiterlily/SoftWarmBeds

jupiterlily committed:
creating path for textures to go in when they're done

jptrrs commented on Defs/Things.xml:
```diff
@@ -217,4 +217,67 @@
+<!--Don't foret to update tex paths when we get the new textures-->
+  <ThingDef ParentName="BeddingBase">
+    <defName>SingleTribalBedding</defName>
+    <label>single tribal bedding set</label>
```
A suggestion for wording: 
I'm a  big fan of using wording to anchor the game thematically. What if we loose the "tribal bedding set" and switched it to "bedding fur"?

jptrrs commented on Defs/Things.xml:
```diff
@@ -217,4 +217,67 @@
+  <ThingDef ParentName="BeddingBase">
+    <defName>SingleTribalBedding</defName>
+    <label>single tribal bedding set</label>
+    <description>A pile of leather blankets with a rudimentary pillow. Not very comfortable but much better than nothing and insulates the user while in bed.</description>
```
Same here: "A pile of fur for one single bed..."

jptrrs commented on Defs/Things.xml:
```diff
@@ -217,4 +217,67 @@
+      <texPath>Things/Beddings/beddingSingle</texPath>
+      <graphicClass>Graphic_Single</graphicClass>
+    </graphicData>
+    <costStuffCount>55</costStuffCount>
```
I kinda feel it should require more pelts than the more sofisticated one. This way, evolving the bedding models would also mean more efficient material use. Just somthing to consider.

jupiterlily replied:
Ah okay, my thinking for reducing the ingredient cost was that they're not having to make a mattress. I don't have strong feelings either way. I suppose the less work to make would offset the higher ingredient cost too.

jptrrs commented on Defs/Things.xml:
```diff
@@ -217,4 +217,67 @@
+      <li>Beddings</li>
+    </thingCategories>
+    <recipeMaker>
+      <researchPrerequisite>CarpetMaking</researchPrerequisite>
```
That's an interesting choice... But I believe it would be more intuitive if it unlocked along the bed that will use it. Rimworld players tend to get confused rather easily.

jupiterlily replied:
So if you're playing with VFEVikings and VFurniture Expanded you'll need to research simple furniture.

This was for the few people who are only playing with Soft Beds and Vikings. From looking at the defs in the viking mod there isn't a prerequisite research for the bed so that'd mean tribal beddings are availiable from the start of the game. I'm not sure how to set it to have no prerequisite, and still have the patch operation that moves it to the VFurniture research.

jptrrs replied:
Hmmm, I see. For a no-prerequisite thing just remove that line. To add it back, it would be a PatchOperationAdd triggered by a PatchOperationFindMod (or in a sequence that already detected the mod).

jptrrs commented on Patches/Patches_VanillaFactionExpandedVikings.xml:
```diff
@@ -8,25 +8,112 @@
+            <fixedStorageSettings>
+              <filter>
+                <categories>
+                  <li>Textiles</li>
```
This would include all textiles, even cloth and wool that ultimately the new beddings can't be made of. Once more in the name of clarity, it would be swell if we could limit the filter to a "leather" category. But I'm not sure if it exists. Gotta check it up.

jupiterlily replied:
I'm confused by this one, I just copied it from the other patches.

Would the following work? My gut says no because I think leathery works like metallic, woody, stony.

<li>Leathery</li>

jptrrs replied:
Yeah, Rimword categories for things are very confusing. There are categories, stuffCategories, thingCategories.... leave it like that, I'll figure it out later.

jupiterlily committed:
lilly's terrible textrues.png

Commit:
Optimized patch.

Commit:
Changed comfort stat to the minimum the game actually uses.

Commit:
Merge branch 'master' of https://github.com/jupiterlily/SoftWarmBeds into pr/18-adding-new-tribal-bedding-type-and-full-compatiblity-for-vfev

jptrrs reviewed (changes requested):
It looks alright at first glance. I made some changes and suggestions, check it out. Will wait for textures to test it all together.

jupiterlily committed:
bedding value tweaks and flavour text

Changed stuffcost from -10% to +20%

Hopefully improved flavour text!

jupiterlily committed:
Merge branch 'master' of https://github.com/jupiterlily/SoftWarmBeds

Commit:
Rounded up flavor text. Looking good!

jupiterlily committed:
Bed and Bedding Textures

jupiterlily commented:
I think we need to make a new def for the bed to change the recipe, right? At the moment it requires wood and cloth wheras it should be just the woody type, right?

jupiterlily committed:
Research modifications

Removed research requirements and updated patch (hopefully correctly)

jptrrs reviewed (approved):
That's it.

jptrrs commented:
> 
> 
> I think we need to make a new def for the bed to change the recipe, right? At the moment it requires wood and cloth wheras it should be just the woody type, right?

No need for a new def, a PatchOperationRemove can fix that.

jupiterlily committed:
Updated Recipes for beds

Kept the wood values the same as in VFEV.

Hopefully formatted it so that it works with mods like Expanded woodworking.

jupiterlily commented:
My artist says that he'll do proper textures for the bedding pelt roll thingys eventually so I just put some rubbish ones together in GIMP as a placeholder. Hope that's okay, I won't be offended if they get replaced or changed in any way.

jupiterlily committed:
Updated texture names to match the code

jptrrs commented on Patches/Patches_VanillaFactionExpandedVikings.xml:
```diff
@@ -8,28 +8,126 @@
+        <xpath>Defs/ThingDef[defName="VFEV_FurBed"]/costStuffCount</xpath>
+        <costStuffCount>45</costStuffCount>
+      </li>
+
```
So... this isn't working. Please, test before submitting. 
You're missing the "value" tags to determine what should be changed. 
Also, if you need to remove one item from a list, it's always best to do just that, instead of replacing the whole list like you're doing on stuffCategories. Think compatibility: if another mod added something to the list and you later rewrite the whole thing, you created an unnecessary incompatibility.

jupiterlily committed:
Fixed red errors

Fixed. Research gating works. Beds are makeable. Stuff amounts work.

jupiterlily committed:
Tweaked Designs

Made beds and masks more rimworldy. kept edges the same so bedding should work with most beds.

Commit:
New bedding texture and changed names for conformity.
